### PR TITLE
[bitnami/mariadb] Allow add secret annotations

### PR DESCRIPTION
**Pull request by juanrgm**
**Description of the change**

Allow add secret annotations.

**Benefits**
Combined with [this](https://github.com/bitnami/charts/pull/3411), we could auto generate passwords from out of chart without helm upgrade conflicts.

**Possible drawbacks**
**Applicable issues**
**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

### CODE REVIEW

File: bitnami/mariadb/templates/secrets.yaml [hunk 1]
```diff
@@ -9,6 +9,12 @@ metadata:
     chart: "{{ template "mariadb.chart" . }}"
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
+  {{- if .Values.secret.annotations }}
+  annotations:
+    {{- range $key, $value := .Values.secret.annotations }}
+    {{ $key }}: {{ $value }}
+    {{- end }}
```
Comment (dani8art) [inline bitnami/mariadb/templates/secrets.yaml:16]: I would use `mysql.tplValues`

```
 annotations: {{- include "mysql.tplValue" (dict "value" .Values.secret.annotations "context" $) | nindent 4 }}
```

File: bitnami/mariadb/values-production.yaml [hunk 2]
```diff
@@ -110,6 +110,13 @@ securityContext:
 ##
 # existingSecret:
 
+## Allow customization of the secret resource
+##
+secret:
+  ## Add custom annotations to the secret
```
Comment (dani8art) [inline bitnami/mariadb/values-production.yaml:116]:
```suggestion
  ## Add custom annotations to the secret (evaluated as a template)
```

File: bitnami/mariadb/values.yaml [hunk 3]
```diff
@@ -110,6 +110,13 @@ securityContext:
 ##
 # existingSecret:
 
+## Allow customization of the secret resource
+##
+secret:
+  ## Add custom annotations to the secret
```
Comment (dani8art) [inline bitnami/mariadb/values.yaml:116]:
```suggestion
  ## Add custom annotations to the secret (evaluated as a template)
```

File: bitnami/mariadb/README.md [hunk 4]
```diff
@@ -81,6 +81,7 @@ The following table lists the configurable parameters of the MariaDB chart and t
...
 | `securityContext.runAsUser`                  | User ID for the container                                                                                                                                                                                                                                                | `1001`                                                            |
 | `existingSecret`                             | Use existing secret for password details (`rootUser.password`, `db.password`, `replication.password` will be ignored and picked up from this secret). The secret has to contain the keys `mariadb-root-password`, `mariadb-replication-password` and `mariadb-password`. | `nil`                                                             |
+| `secret.annotations`                         | Secret annotations                                                                                                                                                                                                                                                       | `{}`                                                              |
```
Comment (dani8art) [inline bitnami/mariadb/README.md:84]:
```suggestion
| `secret.annotations`                         | Secret annotations (evaluated as a template)                                                                                                                                                                                                                                                     | `{}`                                                              |
```

**Comment by juanrgm** [comment]
Hi @dani8art, thanks you for your reply.

The main idea is auto generate the passwords (eg: https://github.com/mittwald/kubernetes-secret-generator) without create an empty secret manually, so with the next config we could install mariadb cleanly without k8s resource files (`secret.yml`).

```yaml
secret:
  requirePasswords: false
  annotations:
    secret-generator.v1.mittwald.de/autogenerate: mariadb-root-password,mariadb-replication-password,mariadb-password
```

File: bitnami/mariadb/templates/secrets.yaml (hunk 1 above)
Comment (juanrgm) [reply bitnami/mariadb/templates/secrets.yaml:16]: `mariadb.tplValues` is not available at [templates/_helpers.tpl](https://github.com/bitnami/charts/blob/master/bitnami/mariadb/templates/_helpers.tpl).

**Pull request merged**